src/Core/Usecase/User/LoginUser.php

```diff
@@ -33,12 +46,32 @@ public function setAuthenticator(PasswordAuthenticator $authenticator)
 		return $this;
 	}
 
+	/**
+	 * @param Flaps $rateLimiter
+	 */
+	public function setRateLimiter(Flap $rateLimiter)
+	{
+		$this->rateLimiter = $rateLimiter;
+	}
+
+	/**
+	 * @param LeakyBucketStrategy $throttlingStrategy
+	 */
+	public function setThrottlingStrategy(LeakyBucketStrategy $throttlingStrategy)
+	{
+		$this->throttlingStrategy = $throttlingStrategy;
+	}
+
 	// Usecase
 	public function interact()
 	{
 		// Fetch the entity, using provided identifiers...
 		$entity = $this->getEntity();
 
+		// Rate limit login attempts
+		$this->rateLimiter->pushThrottlingStrategy($this->throttlingStrategy);
+		$this->rateLimiter->limit($entity->email);
```

Comment:
OK. Similar to the comment above we should be defining our own exception class and throwing that.. otherwise we end up with a dependency on Flaps exception